icicle-vm/src/lib.rs

```diff
@@ -291,13 +291,14 @@ impl Vm {
         }
 
         match self.lift(pc) {
-            Some(group) => {
+            Ok(group) => {
                 self.cpu.block_id = group.blocks.0 as u64;
                 self.cpu.block_offset = 0;
                 VmExit::Running
             }
-            None => {
-                self.cpu.exception = cpu::Exception::new(ExceptionCode::InvalidInstruction, pc);
+            Err(e) => {
+                tracing::trace!("DecodeError at {pc:X}: {e:?}");
```

Comment:
```suggestion
                tracing::trace!("DecodeError at {pc:#x}: {e:?}");
```

(very minor -- everywhere else, the code uses the lowercase hex formatting (i.e. `:#x) so could be changed for consistency)

(Also just fyi, (if you were not already aware), the `#` out the front causes the `0x` prefix to be printed)